src/ru/geekbrains/algorithms/lesson3/queue/DequeImpl.java

```diff
@@ -0,0 +1,57 @@
+package ru.geekbrains.algorithms.lesson3.queue;
+
+import java.util.Arrays;
+
+
+public class DequeImpl<E> extends QueueImpl<E> implements Deque<E> {
+
+/* В QueueImpl изменил модификаторы видимости полей на protected
+ * чтобы они были видны в потомке
+ */
+
+    public DequeImpl(int maxSize) {
+        super(maxSize);
+    }
+
+    @Override
+    public boolean insertLeft(E value) {
+        if (isFull()) {
+            return false;
+        }
+
+        if (head == HEAD_DEFAULT) {
+            head = data.length - 1;
+        }
+        data[head] = value;
+        size++;
+        if (size < data.length) {
+            head--;
+        }
+        return true;
+    }
+
+    @Override
+    public E removeRight() {
+        if (isEmpty()) {
+            return null;
+        }
+
+        if (tail == TAIL_DEFAULT) {
+            ++tail;
```

Comment:
```suggestion
            tail = data.length - 1;
```
иначе мы будем топтаться на одном индексе, если вызывать данный метод несколько раз